src/modal.service.ts

```diff
@@ -0,0 +1,30 @@
+import {Injectable} from '@angular/core';
+
+import {ModalComponent} from './modal.component';
+
+@Injectable()
+export class ModalService {
+  private modals: Array<ModalComponent> = [];
+
+  constructor() {
```

Comment:
Generally, you don't want to want to have listeners nor HTTP calls on your constructor. They are hard to test.

Reply:
thanks for pointing this. Anyway, I will delete the service.